frontend/src/metabase/visualizations/click-actions/drills/mlv2/ColumnFilterDrill.ts

```diff
@@ -0,0 +1,44 @@
+import { t } from "ttag";
+import type { DrillMLv2 } from "metabase/visualizations/click-actions/types";
+import { getColumnFilterDrillPopover } from "metabase/visualizations/click-actions/components/ColumnFilterDrillPopover";
+import * as Lib from "metabase-lib";
+import StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import Filter from "metabase-lib/queries/structured/Filter";
+
+export const ColumnFilterDrill: DrillMLv2<Lib.ColumnFilterDrillThruInfo> = ({
+  drill,
+  drillDisplayInfo,
+  question,
+  clicked,
+}) => {
+  if (!drill || !drillDisplayInfo.initialOp) {
+    return [];
+  }
+
+  const { initialOp } = drillDisplayInfo;
+
+  const query = question.query() as StructuredQuery;
+
+  // TODO: refactor this after Filters will be added to MLv2
+  const initialFilter = new Filter(
```

Comment:
@deniskaber API for adding filters via MLv2 already exist, have you tried using them?

Reply:
Nvm, got it - the popover expects a MLv2 filter